protostar/cheatable_starknet/callable_hint_locals/call_hint_local.py

```diff
@@ -29,19 +32,21 @@ def name(self) -> str:
 
     def _build(
         self,
-    ) -> Callable[[RawAddress, str, Optional[CairoOrPythonData]], CairoOrPythonData]:
+    ) -> Callable[[RawAddress, int, Optional[CairoOrPythonData]], CallResult]:
         return self.call
 
     def call(
         self,
         contract_address: RawAddress,
-        function_name: str,
+        function_selector: int,
         calldata: Optional[CairoOrPythonData] = None,
-    ) -> CairoData:
+    ) -> CallResult:
         return asyncio.run(
             self._call(
                 contract_address=Address.from_user_input(contract_address),
-                entry_point_selector=Selector(function_name),
+                entry_point_selector=Selector(
+                    starknet_keccak(str.encode(short_string_to_str(function_selector)))
```

Comment:
This is because cairo1 compiler uses [`starknet_keccak`](https://github.com/software-mansion-labs/cairo/blob/d903af86ef5fbfa3126861d07c70fa21e2368a26/crates/cairo-lang-starknet/src/contract_class.rs#L214) to calculate entry points' selectors

Reply:
I did `Selector(short_string_to_str(function_name))` instead and it seems more intuitive + uses higher level functions. 
Works the same.

Reply:
Right, for some reason I missed that we get a function name as Cairo short string, thanks for pointing this out.

Reply:
[b5cee22](https://github.com/software-mansion/protostar/pull/1726/commits/b5cee226e668cd3986f0ceaa232c696a7543515b)